lib/osf-components/addon/components/registries/hierarchical-list/index.ts

```diff
@@ -0,0 +1,8 @@
+import NodeModel from 'ember-osf-web/models/node';
+
+export interface HierarchicalListManager {
+    nodesIncludingRoot: NodeModel[];
+    selectedNodes: NodeModel[];
+    onChange: (event: Event, node: NodeModel) => void;
+    isChecked: (node: NodeModel) => boolean;
+}
```

Comment:
Components consuming the `HierarchicalList/item` components should implement this `HierarchcalListManager` class. `PartialRegistrationModal` implements this class.